### PR TITLE
WEB-1233: Load main.js async as well

### PR DESCRIPTION
**Pull request by mlworks**
From the initial double fetch PR, main.js was not downloading until capture.js is done. main.js should download in parallel.

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1233
 **Linked PRs**:
https://github.com/mobify/crabtree-evelyn-progressive/pull/576

**Before**
screen shot 2017-03-07 at 5 51 14 pm

**After**
screen shot 2017-03-07 at 5 49 21 pm

## Changes
- Extract the changes from Crabtree
  - wrap vendor, main, and capture in promises before starting up the app.

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Confirm in Chrome's network tab that main.js is downloading in parallel with capture.js and vendor.js

### CODE REVIEW

**Comment by Bielousov** [comment]
Note: this only works with code splitting in webpack, as well as this change is required for code splitting to work and I don't see it in this PR.

File: web/app/loader.js [hunk 1]
```diff
@@ -131,7 +119,35 @@ if (isReactRoute()) {
             body.appendChild(capturing)
         })
 
-        Promise.all([capturingPromise]).then(() => loadMain(body))
+        const mainPromise = new Promise((resolve) => {
```
Comment (jeremywiebe) [inline web/app/loader.js]: These two blocks look very similar. Do you think it's worth extracting a method to create these script tags?

**Comment by mlworks** [comment]
Ah turns out we have a duplicate PR with the same work. Closing in favor of: https://github.com/mobify/progressive-web-scaffold/pull/414

File: web/app/loader.js [hunk 2]
```diff
@@ -131,7 +119,35 @@ if (isReactRoute()) {
...
+            vendorScript.id = 'progressive-vendor-script'
+            vendorScript.src = getAssetUrl('vendor.js')
+            vendorScript.onload = resolve
+            vendorScript.onerror = resolve
+            body.appendChild(vendorScript)
+        })
+
+         // Load all dependencies asynchronously
+        asyncInitApp()
+        Promise.all([capturingPromise, vendorPromise, mainPromise]).then(asyncRunApp)
```
Comment (jeffkamo) [inline web/app/loader.js]: I accidentally duplicated this work in another PR. @jeremywiebe is this what you mean by returning the result?

https://github.com/mobify/progressive-web-scaffold/pull/414/files#diff-fa44066d20629ee51e62e770692a8900R39

**Pull request closed without merging**